### PR TITLE
refactor: Adjusts search_deployment schema definitions with structure of new schema scaffolding command

### PR DESCRIPTION
**Pull request by AgustinBettati**
## Description

Link to any related issue(s): [CLOUDP-221363](https://jira.mongodb.org/browse/CLOUDP-221363)

The content of the schema is not changed, with the exception of including `Description` and `MarkdownDescription` fields that are generated from description fields in the API Spec.

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [ ] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.

## Further comments

### CODE REVIEW

File: internal/service/searchdeployment/resource_search_deployment_schema.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+	"github.com/hashicorp/terraform-plugin-framework-validators/listvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+func ResourceSchema(ctx context.Context) schema.Schema {
```
Comment (andreaangiolillo) [inline internal/service/searchdeployment/resource_search_deployment_schema.go:16]: Is creating the schema in a separate file a new convention in the repo? Is the what the scaffolding command will do?

File: internal/service/searchdeployment/tfplugingen/generator_config.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+      ignores: []
+
+
+data_sources:
+  search_deployment:
+    read:
+      path: /api/atlas/v2/groups/{groupId}/clusters/{clusterName}/search/deployment
+      method: GET
+    schema:
+      ignores: []
```
Comment (andreaangiolillo) [inline internal/service/searchdeployment/tfplugingen/generator_config.yml]: [nit] Adding a newline at the end of the file
```suggestion
      ignores: []

```

**Review by andreaangiolillo — APPROVED** [review]
LGTM

File: internal/service/searchdeployment/data_source_search_deployment_schema.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:            true,
+				Description:         "Unique 24-hexadecimal digit string that identifies the search deployment.",
+				MarkdownDescription: "Unique 24-hexadecimal digit string that identifies the search deployment.",
+			},
+			"cluster_name": schema.StringAttribute{
+				Required:            true,
+				Description:         "Label that identifies the cluster to return the search nodes for.",
+				MarkdownDescription: "Label that identifies the cluster to return the search nodes for.",
```
Comment (lantoli) [inline internal/service/searchdeployment/data_source_search_deployment_schema.go:21]: will scaffolding be used only for new TF plugin or also for older one?  just to understand if it's enough to generate MarkdownDescription or we also want to generate Description.

or maybe having an input param (defaulted to new plugin?) in the tool if we want to support both?

**Review by lantoli — APPROVED** [review]

File: internal/service/searchdeployment/resource_search_deployment_schema.go (hunk 1 above)
Comment (AgustinBettati) [reply internal/service/searchdeployment/resource_search_deployment_schema.go:16]: correct, scaffolding command will be adjusted to this convention of having the schema and tf models in separate files.

File: internal/service/searchdeployment/tfplugingen/generator_config.yml (hunk 2 above)
Comment (AgustinBettati) [reply internal/service/searchdeployment/tfplugingen/generator_config.yml]: curious about the benefit in ending files with a new line

File: internal/service/searchdeployment/data_source_search_deployment_schema.go (hunk 3 above)
Comment (AgustinBettati) [reply internal/service/searchdeployment/data_source_search_deployment_schema.go:21]: The tool is limited to generating new TF framework code. I have left both definitions as they are both generated by the tool and provide unique value as per comments in the terraform code:

> - MarkdownDescription: used in various tooling, like the documentation generator
> - Description: used in various tooling, like the language server, to give practitioners more information about what this attribute is, what it's for, and how it should be used.

File: internal/service/searchdeployment/tfplugingen/generator_config.yml (hunk 2 above)
Comment (andreaangiolillo) [reply internal/service/searchdeployment/tfplugingen/generator_config.yml]: https://gist.github.com/camh-/1bebfcff1b0f814e9b191edc60d5206b

Comment (lantoli) [reply internal/service/searchdeployment/tfplugingen/generator_config.yml]: +1 ending files with a new line

Comment (AgustinBettati) [reply internal/service/searchdeployment/tfplugingen/generator_config.yml]: Thanks for clarifying. _Configuring vs code settings:_
Screenshot 2024-01-12 at 09 55 31

**Pull request merged**